massa-db-worker/src/massa_db.rs

```diff
@@ -837,3 +850,803 @@ impl MassaDBController for RawMassaDB<Slot, SlotSerializer, SlotDeserializer> {
+            max_history_length: 100,
+            max_new_elements: 100,
+            thread_count: THREAD_COUNT,
+        };
+        let mut db_opts = MassaDB::default_db_opts();
+        // Additional checks (only for testing)
+        db_opts.set_paranoid_checks(true);
+
+        let _db = MassaDB::new_with_options(db_config, db_opts.clone()).unwrap();
+        let db = Arc::new(RwLock::new(
+            Box::new(_db) as Box<(dyn MassaDBController + 'static)>
+        ));
+
+        assert_eq!(
+            Hash::compute_from(db.read().get_xof_db_hash().to_bytes()),
+            initial_hash()
+        );
+
+        // Checks up
+        let cfs = rocksdb::DB::list_cf(&db_opts, temp_dir_db.path()).unwrap_or(vec![]);
+        let cf_exists_1 = cfs.iter().find(|cf| cf == &"state").is_some();
+        let cf_exists_2 = cfs.iter().find(|cf| cf == &"metadata").is_some();
+        let cf_exists_3 = cfs.iter().find(|cf| cf == &"versioning").is_some();
+        println!("cfs: {:?}", cfs);
+        assert_eq!(cf_exists_1, true);
+        assert_eq!(cf_exists_2, true);
+        assert_eq!(cf_exists_3, true);
+        for col in cfs {
+            if col == "metadata" {
+                continue;
+            }
+            assert!(dump_column(db.clone(), col.as_str()).is_empty());
+        }
+
+        // Add data
+
+        let mut batch = DBBatch::new();
+        let b_key1 = vec![1, 2, 3];
+        let b_value1 = vec![4, 5, 6];
+        batch.insert(b_key1.clone(), Some(b_value1.clone()));
+        let mut versioning_batch = BTreeMap::default();
+        let vb_key1 = vec![10, 20, 30];
+        let vb_value1 = vec![40, 50, 60];
+        versioning_batch.insert(vb_key1.clone(), Some(vb_value1.clone()));
+
+        let mut guard = db.write();
+        guard.write_batch(batch.clone(), versioning_batch.clone(), None);
+        drop(guard);
+
+        // assert_eq!(dump_column_opt(db_.clone(), "state"), batch);
+        assert_eq!(dump_column_opt(db.clone(), "state"), batch);
+        // assert_eq!(dump_column_opt(db_.clone(), "versioning"), versioning_batch);
+        assert_eq!(dump_column_opt(db.clone(), "versioning"), versioning_batch);
+
+        // Remove data
+
+        let mut batch_2 = DBBatch::new();
+        batch_2.insert(b_key1.clone(), None);
+        let mut versioning_batch_2 = BTreeMap::default();
+        versioning_batch_2.insert(vb_key1.clone(), None);
+        let mut guard = db.write();
+        guard.write_batch(batch_2.clone(), versioning_batch_2.clone(), None);
+        drop(guard);
+
+        // assert!(dump_column(db_.clone(), "state").is_empty());
+        assert!(dump_column(db.clone(), "state").is_empty());
+        // assert!(dump_column(db_.clone(), "versioning").is_empty());
+        assert!(dump_column(db.clone(), "versioning").is_empty());
+    }
+
+    #[test]
+    fn test_basics_2() {
+        // 1- Init a db + check initial hash
+        // 2- Add some data
+        // 3- Read it
+        // 4- Remove data
```

Comment:
The difference here with the previous test is the delete with prefix? If so, describe it in this comment